api/v2beta2/helmrelease_types.go

```diff
@@ -0,0 +1,1036 @@
+		return InstallRemediation{}
+	}
+	return *in.Remediation
+}
+
+// InstallRemediation holds the configuration for Helm install remediation.
+type InstallRemediation struct {
+	// Retries is the number of retries that should be attempted on failures before
+	// bailing. Remediation, using an uninstall, is performed between each attempt.
+	// Defaults to '0', a negative integer equals to unlimited retries.
+	// +optional
+	Retries int `json:"retries,omitempty"`
+
+	// IgnoreTestFailures tells the controller to skip remediation when the Helm
+	// tests are run after an install action but fail. Defaults to
+	// 'Test.IgnoreFailures'.
+	// +optional
+	IgnoreTestFailures *bool `json:"ignoreTestFailures,omitempty"`
+
+	// RemediateLastFailure tells the controller to remediate the last failure, when
+	// no retries remain. Defaults to 'false'.
+	// +optional
+	RemediateLastFailure *bool `json:"remediateLastFailure,omitempty"`
+}
+
+// GetRetries returns the number of retries that should be attempted on
+// failures.
+func (in InstallRemediation) GetRetries() int {
+	return in.Retries
+}
+
+// MustIgnoreTestFailures returns the configured IgnoreTestFailures or the given
+// default.
+func (in InstallRemediation) MustIgnoreTestFailures(def bool) bool {
+	if in.IgnoreTestFailures == nil {
+		return def
+	}
+	return *in.IgnoreTestFailures
+}
+
+// MustRemediateLastFailure returns whether to remediate the last failure when
+// no retries remain.
+func (in InstallRemediation) MustRemediateLastFailure() bool {
+	if in.RemediateLastFailure == nil {
+		return false
+	}
+	return *in.RemediateLastFailure
+}
+
+// GetStrategy returns the strategy to use for failure remediation.
+func (in InstallRemediation) GetStrategy() RemediationStrategy {
+	return UninstallRemediationStrategy
+}
+
+// GetFailureCount gets the failure count.
+func (in InstallRemediation) GetFailureCount(hr HelmRelease) int64 {
+	return hr.Status.InstallFailures
+}
+
+// IncrementFailureCount increments the failure count.
+func (in InstallRemediation) IncrementFailureCount(hr *HelmRelease) {
+	hr.Status.InstallFailures++
+}
+
+// RetriesExhausted returns true if there are no remaining retries.
+func (in InstallRemediation) RetriesExhausted(hr HelmRelease) bool {
+	return in.Retries >= 0 && in.GetFailureCount(hr) > int64(in.Retries)
+}
+
+// CRDsPolicy defines the install/upgrade approach to use for CRDs when
+// installing or upgrading a HelmRelease.
+type CRDsPolicy string
+
+const (
+	// Skip CRDs do neither install nor replace (update) any CRDs.
+	Skip CRDsPolicy = "Skip"
+	// Create CRDs which do not already exist, do not replace (update) already existing
+	// CRDs and keep (do not delete) CRDs which no longer exist in the current release.
+	Create CRDsPolicy = "Create"
+	// Create CRDs which do not already exist, Replace (update) already existing CRDs
+	// and keep (do not delete) CRDs which no longer exist in the current release.
+	CreateReplace CRDsPolicy = "CreateReplace"
```

Comment:
maybe?
```suggestion
	CreateAndReplace CRDsPolicy = "CreateAndReplace"
```

Reply:
This is copied over from the previous API version, and not worth the change.